src/Exception/DomainException.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Common\Exception;
+
+use DomainException as SplDomainException;
+use InvalidArgumentException;
+use Throwable;
+
+class DomainException extends SplDomainException
+{
+    public const DEFAULT_STATUS_CODE = 500;
+
+    /**
+     * @var string
+     */
+    private $token;
+
+    /**
+     * @var Error[]
+     */
+    private $errors = [];
+
+    public function __construct(
+        string $token,
+        int $statusCode = self::DEFAULT_STATUS_CODE,
+        string $message = ExceptionTokens::AN_ERROR_HAS_OCCURRED,
+        array $errors = [],
+        Throwable $previous = null
+    ) {
+        if (!$this->isExceptionToken($token)) {
+            throw new InvalidArgumentException(ExceptionTokens::INVALID_EXCEPTION_TOKEN, 500);
+        }
+
+        $this->token = $token;
+        $this->errors = $errors;
+
+        parent::__construct($message, $statusCode, $previous);
+    }
+
+    public function getToken(): string
+    {
+        return $this->token;
+    }
+
+    public function getErrors(): array
+    {
+        return $this->errors;
+    }
+
+    private function isExceptionToken(string $token): bool
+    {
+        return preg_match('/[A-Z][A-Z_]/', $token) === 1;
```

Comment:
I'm not so good at regexp but I think this could match things like this: `A text with a UPPERCASE word.`.
https://regex101.com/r/KJmrqI/1